spec/models/customization_template_spec.rb

```diff
@@ -0,0 +1,30 @@
+describe CustomizationTemplate do
+  context "unique name validation" do
+    let(:pxe_image_type)                { FactoryGirl.create(:pxe_image_type) }
+    let(:customization_template)        { FactoryGirl.create(:customization_template, :pxe_image_type => pxe_image_type) }
+    let(:system_customization_template) { described_class.create!(:system => true, :name => "nick") }
+
+    it "doesn't allow same name in same region" do
+      FactoryGirl.create(:customization_template, :name => "fred")
+      expect { FactoryGirl.create(:customization_template, :name => "fred") }.to raise_error(ActiveRecord::RecordInvalid)
+    end
+
+    it "allows different names with the same pxe_image_type in same region" do
+      FactoryGirl.create(:customization_template, :name => "fred", :pxe_image_type => pxe_image_type)
+      FactoryGirl.create(:customization_template, :name => "nick", :pxe_image_type => pxe_image_type)
+      expect(described_class.count).to eq(2)
+    end
+
+    it "allow same name, nil pxe_image_types in different regions (for system seeding)" do
+      ctid = ApplicationRecord.id_in_region(system_customization_template.id, ApplicationRecord.my_region_number + 1)
+      expect { described_class.create!(:system => true, :name => "nick", :id => ctid) }.to change(described_class, :count).by(1)
```

Comment:
I feel like using the `to change()` syntax here would allow this test to succeed even if the first template wasn't created. Can we be more explicit and assert that there are two templates with the same name directly?

Also the fact that the first creation is done in a (lazily evaluated) `let` makes it less clear that it is actually being created.

Reply:
Yeah, that makes sense.  We had this testing the two had the same name originally and I had the brilliant idea of changing it.  @d-m-u was right but didn't stop me.  😉